doc/i2p.md

```diff
@@ -47,9 +52,35 @@ In a typical situation, this suffices:
 bitcoind -i2psam=127.0.0.1:7656
 ```
 
-The first time Bitcoin Core connects to the I2P router, its I2P address (and
-corresponding private key) will be automatically generated and saved in a file
-named `i2p_private_key` in the Bitcoin Core data directory.
+The first time Bitcoin Core connects to the I2P router, if a persistent I2P
+address is to be used, then it (and its corresponding private key) will be
+automatically generated and saved in a file named `i2p_private_key` in the
+Bitcoin Core data directory. A persistent I2P address is used for accepting
+incoming connections (`-i2pacceptincoming`) and for making outgoing connections
+if `-i2ptransientout` is not used.
+
+## Persistent vs transient I2P addresses
+
+In I2P connections, the connection receiver sees the I2P address of the
+connection initiator. This is unlike the Tor network where the recipient does
+not know who is connecting to them, not even the initiator's Tor address.
+
+Even if an I2P node is not accepting incoming connections, it is known to other
+nodes by its outgoing I2P address. If this address is always the same, then
+there is an opportunity to white-list a give node. However, this also creates an
```

Comment:
```suggestion
there is an opportunity to whitelist a given node. However, this also creates an
```

----

Would suggest shortening this and avoiding repeating "a given node" with:

```
Even if an I2P node does not accept incoming connections, it is known to other
nodes by its outgoing I2P address. If this address is always the same when it
connects to other nodes, this creates an opportunity to discriminate,
fingerprint or analyze it based on its I2P address.
```

Reply:
Took the suggestion, but left the "whitelist" text.